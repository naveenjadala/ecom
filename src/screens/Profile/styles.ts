import {StyleSheet} from 'react-native';
import {getFontSize} from '../../assets/fonts/FontSizes';
import {colors} from '../../assets/colors';

export const profileStyles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: '10%',
    backgroundColor: colors.primary,
    marginBottom: '3%',
  },
  profileImage: {
    alignItems: 'center',
    padding: 30,
  },
  profileName: {
    textAlign: 'center',
    color: 'black',
    fontSize: getFontSize(20),
  },
  profileElements: {
    flexDirection: 'row',
  },
  profileEvents: {
    backgroundColor: colors.primary,
  },
});
