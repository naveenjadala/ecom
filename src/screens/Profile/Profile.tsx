import {ScrollView, StyleSheet, Text, View} from 'react-native';
import React, {useState} from 'react';
import ProfileImage from './components/ProfileImage/ProfileImage';
import {profileStyles} from './styles';
import EditButton from './components/EditButton/EditButton';
import SettingsIcon from '../../assets/images/settingIcon.svg';
import PassIcon from '../../assets/images/passIcon.svg';
import OrderIcon from '../../assets/images/orderIcon.svg';
import EventIcon from '../../assets/images/eventIcon.svg';
import ProfileElements from './components/ProfileElements/ProfileElements';
import ProfileEvents from './components/ProfileEvents/ProfileEvents';
import PhotoPickerModal from '../../components/imageUploadComponent/PhotoPickerModal/PhotoPickerModal';
import {Camera, Gallery} from '../../utils/camera';

type Props = {};
/**
 * Profile component renders the user's profile with image, name, and events.
 *
 * @param {Props} props - the props object containing the necessary data
 * @return {JSX.Element} the rendered user profile component
 */
const Profile = (props: Props) => {
  const [openModal, setOpenModal] = useState<boolean>(false);
  const [image, setImage] = useState<string | undefined>('');

  /**
   * Function to open and close the image modal.
   */
  const openImageModal = () => {
    setOpenModal(prevState => !prevState);
  };

  /**
   * Opens the camera and sets the image using the first asset.
   *
   * @return {Promise<void>}
   */
  const openCamera = async () => {
    const camera = await Camera();
    setImage(camera.assets?.[0].uri);
    setOpenModal(false);
  };

  /**
   * Opens the gallery, selects the first image, sets the image, and closes the modal.
   *
   * @return {Promise<void>}
   */
  const openGallery = async () => {
    const camera = await Gallery();
    setOpenModal(false);
    setImage(camera.assets?.[0].uri);
  };

  return (
    <ScrollView>
      <View style={profileStyles.container}>
        <View style={profileStyles.profileImage}>
          <ProfileImage onClick={openImageModal} image={image} />
        </View>
        <Text style={profileStyles.profileName}>Naveen Jadala</Text>
        <View style={profileStyles.profileImage}>
          <EditButton onClick={() => console.log('click')} />
        </View>
        <View style={profileStyles.profileElements}>
          <ProfileElements
            title={'Orders'}
            onClick={() => console.log('click')}
            elementImage={<OrderIcon width={30} height={30} />}
          />
          <ProfileElements
            title={'Pass'}
            onClick={() => console.log('click')}
            elementImage={<PassIcon width={30} height={30} />}
          />
          <ProfileElements
            title={'Events'}
            onClick={() => console.log('click')}
            elementImage={<EventIcon width={30} height={30} />}
          />
          <ProfileElements
            title={'Settings'}
            onClick={() => console.log('click')}
            elementImage={<SettingsIcon width={30} height={30} />}
          />
        </View>
      </View>
      <View style={profileStyles.profileEvents}>
        <ProfileEvents
          title="INBOX"
          subTitle="View message"
          onClick={() => console.log('click')}
        />
        <ProfileEvents
          title="YOUR NIKE MEMBER REWARDS"
          subTitle="View rewards"
          onClick={() => console.log('click')}
        />
      </View>
      <PhotoPickerModal
        visible={openModal}
        closeModal={openImageModal}
        openCamera={openCamera}
        openGallery={openGallery}
      />
    </ScrollView>
  );
};

export default Profile;

const styles = StyleSheet.create({});
