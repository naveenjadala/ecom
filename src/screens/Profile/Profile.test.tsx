import {fireEvent, render, screen, waitFor} from '../../test/test-utils';
import Profile from './Profile';
import {launchCamera, launchImageLibrary} from 'react-native-image-picker';

jest.mock('react-native-image-picker', () => ({
  launchCamera: jest.fn(),
  launchImageLibrary: jest.fn(),
}));

// Mock the Camera utility separately
jest.mock('../../utils/camera', () => ({
  Camera: jest.fn().mockReturnValue({
    assets: [{uri: 'mocked_camera_image_uri'}],
  }),
  Gallery: jest.fn().mockReturnValue({
    asserts: [{uri: 'mocked_gallery_image_uri'}],
  }),
}));

describe('Profile', () => {
  test('should render Profile screen snapShort', () => {
    render(<Profile />);
    expect(screen).toMatchSnapshot();
  });
});

describe('should render Profile', () => {
  test('should render photo modal on click', () => {
    render(<Profile />);

    // Check if photo modal is not open
    const getPhotoModal = screen.queryByTestId('photoPickerModal');
    expect(getPhotoModal).toBeNull();

    // Open photo modal
    fireEvent.press(screen.getByTestId('profileImage'));
    const getPhotoModalAfterClick = screen.queryByTestId('photoPickerModal');
    expect(getPhotoModalAfterClick).toBeTruthy();

    // Close photo modal
    fireEvent.press(screen.getByTestId('cancel'));
    const getPhotoModalAfterClose = screen.queryByTestId('photoPickerModal');
    expect(getPhotoModalAfterClose).toBeNull();
  });

  test('should open camera', async () => {
    render(<Profile />);
    // Check if the modal is visible
    fireEvent.press(screen.getByTestId('profileImage'));
    const getPhotoModalAfterClick = screen.queryByTestId('photoPickerModal');
    expect(getPhotoModalAfterClick).toBeTruthy();

    // Open camera
    const getCameraButton = screen.getByTestId('takePhoto');
    fireEvent.press(getCameraButton);
    await waitFor(() =>
      expect(screen.queryByTestId('photoPickerModal')).toBeNull(),
    );

    // Check if image is set
    await waitFor(() =>
      expect(screen.getByTestId('img').props.source.uri).toBe(
        'mocked_camera_image_uri',
      ),
    );
  });
});
