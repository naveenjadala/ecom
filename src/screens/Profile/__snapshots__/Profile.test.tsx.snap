// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render Profile screen snapShort 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "minHeight": "100%",
      "minWidth": "100%",
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "marginBottom": "3%",
            "paddingTop": "10%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "padding": 30,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "borderRadius": 100,
                "height": 100,
                "justifyContent": "center",
                "opacity": 1,
                "width": 100,
              }
            }
            testID="profileImage"
          >
            <test-file-stub
              height={100}
              width={100}
            />
          </View>
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 10,
              "textAlign": "center",
            }
          }
        >
          Naveen Jadala
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "padding": 30,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "borderColor": "#E4E4E4",
                "borderRadius": 5,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 6,
                "width": "60%",
              }
            }
          >
            <Text
              style={{}}
            >
              Edit Profile
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <test-file-stub
              height={30}
              width={30}
            />
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 6,
                  "paddingVertical": 10,
                }
              }
            >
              Orders
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <test-file-stub
              height={30}
              width={30}
            />
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 6,
                  "paddingVertical": 10,
                }
              }
            >
              Pass
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <test-file-stub
              height={30}
              width={30}
            />
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 6,
                  "paddingVertical": 10,
                }
              }
            >
              Events
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <test-file-stub
              height={30}
              width={30}
            />
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 6,
                  "paddingVertical": 10,
                }
              }
            >
              Settings
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#FFFFFF",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingHorizontal": "10%",
              "paddingVertical": "5%",
            }
          }
        >
          <View>
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 8,
                  "letterSpacing": 0,
                  "textAlignVertical": "center",
                }
              }
            >
              INBOX
            </Text>
            <Text
              style={
                {
                  "color": "#8C8C8C",
                  "fontSize": 6.5,
                }
              }
            >
              View message
            </Text>
          </View>
          <test-file-stub
            height={20}
            style={
              {
                "alignSelf": "center",
              }
            }
            width={20}
          />
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingHorizontal": "10%",
              "paddingVertical": "5%",
            }
          }
        >
          <View>
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 8,
                  "letterSpacing": 0,
                  "textAlignVertical": "center",
                }
              }
            >
              YOUR NIKE MEMBER REWARDS
            </Text>
            <Text
              style={
                {
                  "color": "#8C8C8C",
                  "fontSize": 6.5,
                }
              }
            >
              View rewards
            </Text>
          </View>
          <test-file-stub
            height={20}
            style={
              {
                "alignSelf": "center",
              }
            }
            width={20}
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
